hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -221,11 +223,15 @@ public void notifyCheckpointComplete(long checkpointId) {
           // the stream write task snapshot and flush the data buffer synchronously in sequence,
           // so a successful checkpoint subsumes the old one(follows the checkpoint subsuming contract)
           final boolean committed = commitInstant(this.instant, checkpointId);
+
+          if (tableState.scheduleCompaction) {
+            // if async compaction is on, schedule the compaction
+            metaClient.reloadActiveTimeline();
```

Comment:
can we move these 3 lines into the `CompactionUtil#scheduleCompaction(metaClient)` ?